### PR TITLE
Add content for the Using Cobra page

### PR DESCRIPTION
**Pull request by mjeronimo**
Signed-off-by: Michael Jeronimo <michael.jeronimo@openrobotics.org>

### CODE REVIEW

File: source/How-To-Guides/Use-Cobra.rst [hunk 1]
```diff
@@ -1,7 +1,351 @@
...
+.. code-block::
+
+    [
+    { "type"      :       "Fct names also used as variables",
+        "message"   :       "lines 688..688",
+        "file"      :       "/home/spaceros-user/spaceros/src/rcutils/src/logging.c",
+        "line"      :       688,
+        "cobra"     :       "1 10 0"
+    }
+    { "type"      :       "Fct names also used as variables",
```
Comment (nuclearsandwich) [inline source/How-To-Guides/Use-Cobra.rst:102]: Is this file a single JSON array, in which case there should probably be a comma here, or a sequence of individual JSON objects, in which case the opening square bracket is probably incorrect?

**Review by nuclearsandwich — APPROVED** [review]
One note about the example JSON formatting but otherwise it looks good!

File: source/How-To-Guides/Use-Cobra.rst (hunk 1 above)
Comment (mjeronimo) [reply source/How-To-Guides/Use-Cobra.rst:102]: Thanks, I didn't notice that. As a test, I just tried the nightly Docker image and cobra indeed outputs this file without the comma (an invalid JSON format). It's a bit weird as the file output by cobra has a .txt filename extention, but it's supposed to be a JSON file. The .txt file is then read by their json_convert utility to generate SARIF. The utility reads the file OK and generates valid SARIF. 

So, I can think of a couple options:
* Use the term "JSON-like" in the docs and leave the doc as-is
* Update the doc to include the comma and enter an issue on the Cobra repo

Thoughts?

Comment (nuclearsandwich) [reply source/How-To-Guides/Use-Cobra.rst:102]: > Thoughts?

Oof.

So a "stream" of JSON documents is equally valid contents. But I think if the file contains opening and closing square brackets / braces (`[]`) but does not put commas between objects then it is neither a stream of multiple JSON documents nor a single valid JSON array and that's probably worth ticketing upstream.

In the meantime, our documentation should accurately reflect the current reality, whether that be a single JSON document, a sequence of them, or some almost-JSON format.

Comment (mjeronimo) [reply source/How-To-Guides/Use-Cobra.rst:102]: Ok, thanks. I'll ticket that upstream and keep the docs as-is to reflect the current state of things. Maybe introduce "JSON-like" at that point in the doc.

**Pull request merged**